airflow/models/dag.py

```diff
@@ -684,8 +678,11 @@ def __init__(
             self.timetable = DatasetTriggeredTimetable(DatasetAll(*schedule))
             self.schedule_interval = self.timetable.summary
         elif isinstance(schedule, ArgNotSet):
-            self.timetable = create_timetable(schedule, self.timezone)
-            self.schedule_interval = DEFAULT_SCHEDULE_INTERVAL
+            if airflow_conf.getboolean("scheduler", "default_none_schedule"):
+                self.schedule_interval = None
+            else:
+                self.schedule_interval = timedelta(days=1)
```

Comment:
We should raise a deprecation warning here, including instructions what to do.

Reply:
(that config is disabled in main so it won't generate the warnings for examples)